rfcs/20220201-quantization-scaling/README.md

```diff
@@ -0,0 +1,264 @@
+Proposal for updating quantization scale functionality in oneDNN
+================================================================
+
+## 0. Motivation
+
+There are multiple requests for extending scaling support in oneDNN primitives:
+- run-time scaling in Eltwise post-op;
+- scaling for src1 in Binary post-op;
+- scaling for inputs in Concat.
+
+All of these requests can be addressed separately, but taking into account
+already existing oneDNN mechanisms can help to avoid duplications and to unify
+scaling in the future.
+
+
+## 1. Introduction
+
+This chapter provides a comprehensive overview of scaling support in oneDNN.
+
+
+### 1.1. Quantization
```

Comment:
How do we plan to position scales in the library? I remember we wanted to define scales solely for INT8 primitives to support quantization. But now we seem to be generalizing them. Do I get it right that the long-term plan is to provide a unified scaling capability that could be used either for quantization or general purposes?

Reply:
We want to support them in case of mixed data type (f16/bf16/f32<->s8) and general int8 flow. Other than that we will probably have to support it in binary for all types for solver implementation.

Reply:
btw, matmul also seems to be supporting output scales for non-int8 types.

Reply:
The scales will be supported for non-int8 as well, since we want to replace op desc scales parameter.